plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,735 @@
+    auth_password = module.params.get("auth_password")
+    ebgp_multi_hop_ttl = module.params.get("ebgp_multi_hop_ttl")
+    site_of_origin = module.params.get("site_of_origin")
+    weight = module.params.get("weight")
+    allowed_self_as_count = module.params.get("allowed_self_as_count")
+    local_asn_config = LOCAL_ASN_CONFIG.get(module.params.get("local_asn_config"))
+    local_asn = module.params.get("local_asn")
+    import_route_map = module.params.get("import_route_map")
+    export_route_map = module.params.get("export_route_map")
+    peer_prefix = module.params.get("peer_prefix")
+    bgp_controls = module.params.get("bgp_controls")
+    peer_controls = module.params.get("peer_controls")
+    address_families = module.params.get("address_families")
+    private_as_controls = module.params.get("private_as_controls")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "l3out", template)
+    mso_template.validate_template("l3out")
+
+    tenant_id = mso_template.template_summary.get("tenantId")
+    tenant_name = mso_template.template_summary.get("tenantName")
+
+    l3out_object = mso_template.get_l3out_object(name=l3out, fail_module=True)
+    node_group_object = mso_template.get_l3out_node_group(node_group, l3out_object.details, fail_module=True)
+    bgp_peers = node_group_object.details.get("bgpPeers", [])
+
+    bgp_peer_objects = get_bgp_peer_by_address(mso_template, bgp_peers, ipv4_addr=ipv4_addr, ipv6_addr=ipv6_addr)
+
+    if bgp_peer_objects and (ipv4_addr or ipv6_addr):
+        set_bgp_peer_relations_name(mso, bgp_peer_objects.details)
+        mso.existing = mso.previous = copy.deepcopy(bgp_peer_objects.details)  # Query a specific object
+    elif bgp_peer_objects:
+        mso.existing = [set_bgp_peer_relations_name(mso, bgp_peer) for bgp_peer in bgp_peer_objects]  # Query all objects
+
+    if state != "query":
+        bgp_peer_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}/bgpPeers/{2}".format(
+            l3out_object.index, node_group_object.index, bgp_peer_objects.index if bgp_peer_objects else "-"
+        )
+
+    ops = []
+
+    if state == "present":
+        route_map_objects = []
+        if import_route_map or export_route_map:
+            route_map_objects = mso.query_objs(generate_api_endpoint("templates/objects", **{"type": "routeMap", "tenant-id": tenant_id}))
+
+        peer_prefix_uuid = None
+        if peer_prefix:
+            peer_prefix_objects = mso.query_objs(generate_api_endpoint("templates/objects", **{"type": "bgpPeerPrefixPol", "tenant-id": tenant_id}))
+            peer_prefix_uuid = get_peer_prefix_object(mso_template, peer_prefix_uuid, peer_prefix, peer_prefix_objects).details.get("uuid")
+
+        mso_values = dict(
+            peerAddressV4=ipv4_addr,
+            peerAddressV6=ipv6_addr,
+            peerAsn=remote_asn,
+            adminState=admin_state,
+            authEnabled=True if auth_password else None,
```

Comment:
so false is not a valid option?

Reply:
False is valid option, when the `auth_password` is None, `authEnabled` sets to `false` by default.